agent/src/test/java/com/teamscale/jacoco/agent/testimpact/TestExecutionWriterTest.java

```diff
@@ -0,0 +1,44 @@
+package com.teamscale.jacoco.agent.testimpact;
+
+import com.teamscale.report.testwise.model.ETestExecutionResult;
+import com.teamscale.report.testwise.model.TestExecution;
+import org.junit.jupiter.api.Test;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class TestExecutionWriterTest {
+
+	@Test
+	public void testOneExecution() throws Exception {
+		Path tempFile = Files.createTempFile("TestExecutionWriterTest", "tmp");
+		try {
+			TestExecutionWriter writer = new TestExecutionWriter(tempFile.toFile());
+			writer.append(new TestExecution("test1", 123, ETestExecutionResult.PASSED));
+			String json = String.join("\n", Files.readAllLines(tempFile));
+			assertThat(json).isEqualTo("[{\"durationMillis\":123,\"result\":\"PASSED\",\"uniformPath\":\"test1\"}]");
+		} finally {
+			tempFile.toFile().delete();
+		}
+	}
+
+	@Test
+	public void testMultipleExecutions() throws Exception {
```

Comment:
```suggestion
	public void testMultipleExecutions(@TempDir File tmp) throws Exception {
```

The folder is also automatically removed after the test. So no need for explicit cleanup